### DISCUSSION TITLE
Uploading images via API

### DISCUSSION THREAD

**Discussion by bmachek** [open-webui/open-webui#7205] [Issues]
Hi there,

is it yet supported to upload images via the API endpoint /api/v1/files?
If I try, I get an error 

```RuntimeError: Could not detect encoding for /app/backend/data/uploads/31db174b-20a7-4f4d-8508-e7a1b9d01d7c__6_C0243.jpg```

Looking at backend/open_webui/apps/retrieval/loaders/main.py it seems there is no loader for JPEG.

Is there a way around that?

If I use the interactive webui, uploading the same image is working.

Many thanks in advance!
Bastian

**Comment by tjbck** [comment]
You'll have to encode the image to base64 and attach as base64 encoded image for now, hope that helps!

**Reply by bmachek** [reply]
Yep, solved it. Many thanks. :-)

**Reply by bmachek** [reply]
Sorry, have a followup question. If I reference file in a chat:
```
curl -X POST http://localhost:3000/api/chat/completions \
-H "Authorization: Bearer xxx" \
-H "Content-Type: application/json" \
-d '{
      "model": "llama3.2-vision:latest",
      "messages": [
        {"role": "user", "content": "describe the image contents"}
      ],
      "files": [
        {"type": "file", "id": "909ae159-5d79-4ddc-af74-aed43b0a4024"}
      ]
    }'
```

I get the response:
```
{"id":"llama3.2-vision:latest-999ffd73-c00c-4804-827b-b6ad3f2f2309","created":1732267380,"model":"llama3.2-vision:latest","choices":[{"index":0,"logprobs":null,"finish_reason":"stop","message":{"content":"A fun task!\n\nAfter analyzing the provided XML data, I'll attempt to describe the contents of the images. Please note that this is a text-based analysis and might not be 100% accurate.\n\nThe XML data contains three blocks of encoded image data. Each block has a `<source>` tag with a `<source_context>` tag containing a base64-encoded string representing the image data. I'll decode these strings to extract the actual image data.\n\n**Image 1:**\nDecoded from <source>...</source>, this image appears to be a text document or a webpage. The decoded image data contains HTML code, including links and images, indicating that it's likely an extracted snapshot of a web page.\n\n**Image 2:**\nThis image seems to be a screenshot of a chat window or a conversation interface. The decoded data contains various emoticons, emojis, and text strings, suggesting that the image is a representation of a chat session.\n\n**Image 3:**\nThe third image appears to be a graphical user interface (GUI) element, possibly from an application or software. The decoded data contains elements like buttons, menus, and other UI components, indicating that it's likely a screenshot of a desktop application or a software window.\n\nPlease note that these descriptions are based on the text-based analysis of the encoded image data and might not accurately represent the actual contents of the images.\n\nIf you have any further questions or would like me to clarify anything, feel free to ask!","role":"assistant"}}],"object":"chat.completion"}%    
```


Do I have to specify a content-type at upload or some other point?


Many thanks again!

**Comment by Sapphire38** [comment]
@bmachek can you put the example here please?

**Reply by bmachek** [reply]
I moved to using the Ollama API directly. I'm sorry I don't have the example still available, the code is gone, and I don't remember which picture I used for testing. Thanks anyways. :-)

**Comment by evanrsl** [comment]
also have the same question here. is there any working example?

**Comment by thistleknot** [comment]
curious
how does using docling as the content Extraction Engine (within document tab) handle when docling converts items to base64?

I haven't tried docling directly from docker, idk if it even by default attempts to extract images.  But it's a new feature I've noticed, and if images are supported, how are they chunked and sent with other information?  i would imagine they would never be chunked because that would break the image inference.  Can someone explain how this logic is handled (if it is even handled at all)?
